crates/core/src/host/module_host.rs

```diff
@@ -71,31 +69,25 @@ impl DatabaseUpdate {
         false
     }
 
-    pub fn from_writes(stdb: &RelationalDB, tx_data: &TxData) -> Self {
-        let mut map: HashMap<TableId, (Vec<ProductValue>, Vec<ProductValue>)> = HashMap::new();
-        for record in tx_data.records.iter() {
-            let pv = record.product_value.clone();
-            let table = map.entry(record.table_id).or_default();
-            match record.op {
-                TxOp::Delete => &mut table.0,
-                TxOp::Insert(_) => &mut table.1,
-            }
-            .push(pv);
+    pub fn from_writes(tx_data: &TxData) -> Self {
+        let mut map: HashMap<TableId, DatabaseTableUpdate> = HashMap::new();
```

Comment:
TODO: this changed since last rebase, we should be able to just clone the `Arc<[ProductValue]>` into the `DatabaseTableUpdate`.

Reply:
Changed `DatabaseTableUpdate` to contain `Arc[ProductValue]>` instead of `Vec<ProductValue>` as of [7065dad5bbca2c5271fab0705d7b8f0196eab1d5](https://github.com/clockworklabs/SpacetimeDB/compare/a5bd86d3f24a2db0a188514fd39c78492be49e57..7065dad5bbca2c5271fab0705d7b8f0196eab1d5)

cc @Centril as you have touched this code most recently

(Added missing #[inline] in [db0ff831c2540973f00c41d31ea6f4455d904614](https://github.com/clockworklabs/SpacetimeDB/compare/7065dad5bbca2c5271fab0705d7b8f0196eab1d5..db0ff831c2540973f00c41d31ea6f4455d904614))